include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -298,18 +298,14 @@ struct reduce_over_group
         auto __local_idx = __item_id.get_local_id(0);
         auto __group_size = __item_id.get_local_range().size();
 
-        auto __k = 1;
-
-        do
+        for (::std::uint32_t __k = 1; __k < __group_size; __k <<= 1)
```

Comment:
Let's use `__k *= 2` here. Compilers know how to optimize it, and the code would be more self-descriptive. We can also use a more descriptive name, e.g. `__power_2`

Reply:
I will change this.